### PR TITLE
Add JWT authenticator

### PR DESCRIPTION
**Pull request by isharak**
## Purpose
This module will add JWT authenticator  and http jwt authentication handler.

### CODE REVIEW

**Comment by anupama-pathirage** [comment]
@isharak  : There are test failures in this PR. Can you please check?

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/auth/jwt/JWTAuthenticatorTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    private static final String AUDIENCE = "audience";
+    private static final String CERTIFICATE_ALIAS = "certificateAlias";
+    private static final String JWT_AUTH_CACHE_CONFIG = "jwt_auth_cache";
+    private static final String ENABLED = "enabled";
+    private static final String EXPIRY_TIME = "expiryTime";
+    private static final String CAPACITY = "capacity";
+    private static final String EVICTION_FACTOR = "evictionFactor";
+
+    private ConfigRegistry initialConfigRegistry;
+    private String resourceRoot;
```
Comment (anupama-pathirage) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/auth/jwt/JWTAuthenticatorTest.java]: Shall we convert this to local variable?

File: stdlib/ballerina-builtin/src/main/ballerina/ballerina/jwt/natives.bal [hunk 2]
```diff
@@ -17,6 +17,8 @@
 package ballerina.jwt;
 
 //JOSH header parameters
+
+const string TMP = "tmp";
```
Comment (anupama-pathirage) [inline stdlib/ballerina-builtin/src/main/ballerina/ballerina/jwt/natives.bal]: Are we using this?

**Comment by isharak** [comment]
Remove the logs and took that to the handler level which will be replaced with interceptor so we can propagate that to client

**Pull request merged**